workers/unity/Packages/io.improbable.worker.sdk/IOStream.cs

```diff
@@ -26,95 +32,104 @@ public static IOStream CreateRingBufferStream(uint capacity)
             return new IOStream(CIO.CreateRingBufferStream(capacity));
         }
 
-        public static IOStream CreateFileStream(string fileName)
+        public static IOStream CreateFileStream(string fileName, OpenMode openMode)
         {
             fixed (byte* fileNameBytes = ApiInterop.ToUtf8Cstr(fileName))
             {
-                return new IOStream(CIO.CreateFileStream(fileNameBytes, CIO.OpenMode.OpenModeDefault));
+                return new IOStream(CIO.CreateFileStream(fileNameBytes, (CIO.OpenMode) openMode));
```

Comment:
I'm not sure if this kind of cast is actually valid. the C# Interop has some fancy conversion class for this reason.
Might want to double check

Reply:
It would be a compiler error if it was an invalid cast I think (the `as` casts fail at runtime)